### ISSUE TITLE
Adaptive theme mode doesn't automatically switch the theme when changed

### ISSUE THREAD

**Issue by d3vhound** [jpudysz/react-native-unistyles#471]
### Description

So I'm setting up a new project with unistyles v3 beta 4 and I've noticed when I change the theme via phone settings and havee adaptive theme enabled it doesn't automatically switch unless I reload the app. I guess this is a new bug introduced because it worked before.

### Steps to Reproduce

1. implement unistyles v3 beta 4 in new expo app with adaptive themes set up
2. Run app
3. Switch phone theme 
4. Check if theme changed

### Unistyles Version

3.0.0-beta4

### React Native Version

76

### Platforms

iOS

### Expo

Yes

**Comment by prescarlton** [comment]
Also seeing this issue, although very intermittently. I've been trying to track down exactly when it does / does not switch but haven't been able to pinpoint it yet.

**Comment by d3vhound** [comment]
@jpudysz just a gentle reminder

**Comment by jpudysz** [comment]
Will check it before releasing `beta.5`

**Comment by jpudysz** [comment]
Not sure how to repro, works for me. Maybe it's a babel issue and your components are not detected? 

https://github.com/user-attachments/assets/47f2fa4d-2b80-424c-a9a4-9d37d522a4ff

**Comment by d3vhound** [comment]
Interesting, my babel.config.js is just a replica of the doc example: 
```js
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: [
      ['react-native-unistyles/plugin']
  ]
  };
};
```

Could it be something else?

**Comment by jpudysz** [comment]
It's really hard to say. Without a repro I can't see the issue. 

Can you repro it in fresh starter or can you isolate the issue to single screen and show me the code?

**Comment by d3vhound** [comment]
Ah I think it might be related to this warning I've been getting in terminal after running the app:

```bash
The package /Users/devionvillegas/Developer/Oshi/oshi-mobile-monorepo/node_modules/react-native-unistyles contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/devionvillegas/Developer/Oshi/oshi-mobile-monorepo/node_modules/react-native-unistyles" defined in "exports" is /Users/devionvillegas/Developer/Oshi/oshi-mobile-monorepo/node_modules/react-native-unistyles/lib/commonjs/index, however this file does not exist. Falling back to file-based resolution.
```

**Comment by jpudysz** [comment]
Can you see this file on disk though?

**Comment by d3vhound** [comment]
Yeah it exists. Could it be my metro.config.js and the fact I'm using turborepo?

```js
// Learn more https://docs.expo.dev/guides/monorepos
const { getDefaultConfig } = require('expo/metro-config');
const path = require('path');

const projectRoot = __dirname;

const config = getDefaultConfig(projectRoot);

config.watchFolders = [path.join(__dirname, '..', '..')];
config.resolver.unstable_enableSymlinks = true;
config.resolver.unstable_enablePackageExports = true;

module.exports = config;
```

**Comment by Brentlok** [comment]
Can you try adding `nohoist`? https://classic.yarnpkg.com/blog/2018/02/15/nohoist/#how-to-use-it

**Comment by d3vhound** [comment]
@Brentlok I'm using a pnpm turborepo setup. Is yarn preferred over pnpm?

**Comment by jpudysz** [comment]
How pnpn scopes node_modules? You have a single node_modules in the root folder or one per workspace ?

**Comment by d3vhound** [comment]
Yeah single node_modules in root folder.

**Comment by jpudysz** [comment]
Will try to repro with local pnpm monorepo.

**Comment by jpudysz** [comment]
I downloaded https://github.com/byCedric/expo-monorepo-example that uses pnpm monorepo.
Installed Unistyles, Nitro 0.18.2 and replaced `App.tsx` with:

```tsx
import { HomeIcon, HomeScreen } from '@acme/feature-home';
import { StatusBar } from 'expo-status-bar';
import { View } from 'react-native';
import { StyleSheet } from 'react-native-unistyles';
import './unistyles'

export default function App() {
  return (
    <View style={styles.container}>
      <HomeIcon style={{ fontSize: 64 }} />
      <HomeScreen />
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create((theme) => ({
  container: {
    flex: 1,
    backgroundColor: theme.colors.backgroundColor,
    alignItems: 'center',
    justifyContent: 'center',
  },
}))

```

Result:

https://github.com/user-attachments/assets/71a40dad-2896-4ced-883d-e79af6980842

Everything works great, so @d3vhound, unless you provide a repro, I'm unable to help.

**Comment by d3vhound** [comment]
Interesting. I think you just help confirm for me that it might be expo-router related. I'll try to setup a repro and link it here shortly.

**Comment by jpudysz** [comment]
Thanks!

**Comment by jpudysz** [comment]
Hey, we released `.beta5`. Can you check it again?

**Comment by jpudysz** [comment]
That's not the solution 😁. Can you provide a repro?

From the code snippet above I can also see an issue here:

```tsx
const ButtonStyle = StyleSheet.create(theme => ({
  button: theme.components.button,
  buttonText: theme.components.buttonText,
}))
```

Babel won't detect your dependencies + this is anti-pattern. You're copying button related styles across all stylesheets and components. Please check this section: https://www.unistyl.es/v3/other/babel-plugin#1-detecting-stylesheet-dependencies

**Comment by jpudysz** [comment]
Added support for this inline syntax:

```ts
const ButtonStyle = StyleSheet.create(theme => ({
  button: theme.components.button,
  buttonText: theme.components.buttonText,
}))
```
